File: Algorithms/1991. Find the Middle Index in Array/golang/solution.go
func findMiddleIndex(nums []int) int {
	var (
		left  int
		right int
	)
	for _, val := range nums {
		right += val
	}

	for i, val := range nums {
		right -= val
		if left == right {
			return i
		}
		left += val
	}
	return -1
}

File: Algorithms/1991. Find the Middle Index in Array/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findMiddleIndex($nums)
    {
        $left = 0;
        $right = array_sum($nums);
        foreach ($nums as $i => $value) {
            $right -= $value;
            if ($left == $right) {
                return $i;
            }
            $left += $value;
        }

        return -1;
    }
}
